secret/secret_test.go

```diff
@@ -305,17 +283,16 @@ func TestGenerateSecretsWithNoSecrets(t *testing.T) {
 			},
 		}
 
-		secrets := v1.Secret{Data: map[string][]byte{}}
-		updates := v1.Secret{Data: map[string][]byte{}}
+		secrets := &v1.Secret{Data: map[string][]byte{}}
+		updates := &v1.Secret{Data: map[string][]byte{}}
 
 		var mockSecrets MockSecrets
 		generateSSLCerts = mockSecrets.GenerateSSLCerts
 		passGenerate = mockSecrets.PassGenerate
-		mockSecrets.On("GenerateSSLCerts", &secrets, &updates).Return(false)
-		mockSecrets.On("PassGenerate", map[string][]byte{}, map[string][]byte{}, "dirty").Return(true)
-		dirty := GenerateSecrets(manifest, &secrets, &updates)
-		assert.True(dirty)
-		mockSecrets.AssertCalled(t, "PassGenerate", map[string][]byte{}, map[string][]byte{}, "dirty")
+		mockSecrets.On("GenerateSSLCerts", secrets, updates)
+		mockSecrets.On("PassGenerate", secrets, updates, "dirty")
+		GenerateSecrets(manifest, secrets, updates)
+		mockSecrets.AssertCalled(t, "PassGenerate", secrets, updates, "dirty")
```

Comment:
Assert that the secrets are dirty?

Reply:
They won't be because we have mocked `PassGenerate`, which would be setting the dirty flag. This is a test for `GenerateSecrets` which doesn't touch the `dirty` flag, so there is no reason to test for it here. Same for the next comment.